In [1]:
from src.retrievals.sparse import KomoranRetrieval, OktRetrieval, TokenizerRetrieval, ElasticSearchRetrieval

In [2]:
from datasets import load_from_disk

In [3]:
datasets = load_from_disk("../data/new_train_doc")

In [4]:
retrieval = ElasticSearchRetrieval(data_path='../data')
retrieval.get_sparse_embedding()

In [5]:
from tqdm import tqdm

k = 10
score = 0
correct_examples = []
wrong_set = []
for i, data in enumerate(tqdm(datasets['validation'])):
  query = data['question']
  # query = " ".join(map(lambda x: x[0], filter(lambda x: x[1] == 'NNP' or 'NNG', mecab.pos(query))))
  scores, retrieved_examples = retrieval.get_relevant_doc(query, k=k)
  if data['context'] in retrieved_examples:
    correct_examples.append(retrieved_examples)
    score += 1
  else: 
    wrong_set.append(data)

  0%|          | 0/841 [00:00<?, ?it/s]/home/ubuntu/workspace/mrc_venv/lib/python3.8/site-packages/elasticsearch/connection/base.py:209: ElasticsearchWarning: Elasticsearch built-in security features are not enabled. Without authentication, your cluster could be accessible to anyone. See https://www.elastic.co/guide/en/elasticsearch/reference/7.15/security-minimal-setup.html to enable security.
  warnings.warn(message, category=ElasticsearchWarning)
100%|██████████| 841/841 [00:22<00:00, 36.75it/s]


In [6]:
score/len(datasets['validation']) * 100

89.53626634958383

- 코모란 : top10 88.46611177170035
- Okt : top10 90.24970273483947
- xlm-roberta-large : top10 74.19738406658739
- klue-roberta-large : top10 87.63376932223544
- koelectra-v3 : top10 87.99048751486326
- elasticsearch : top10 89.53626634958383